src/topology/algebra/module/finite_dimension.lean

```diff
@@ -366,6 +370,36 @@ by { ext x, refl }
   f.to_continuous_linear_equiv_of_det_ne_zero hf x = f x :=
 rfl
 
+lemma to_lin_prod_continuous_linear_map (a b c d : 𝕜) :
+  (matrix.to_lin (basis.fin_two_prod 𝕜) (basis.fin_two_prod 𝕜)
+      ![![a, b], ![c, d]]).to_continuous_linear_map =
+  (a • continuous_linear_map.fst 𝕜 𝕜 𝕜 + b • continuous_linear_map.snd 𝕜 𝕜 𝕜).prod
+  (c • continuous_linear_map.fst 𝕜 𝕜 𝕜 + d • continuous_linear_map.snd 𝕜 𝕜 𝕜) :=
+begin
+  ext;
+  simp only [continuous_linear_map.coe_comp', linear_map.coe_to_continuous_linear_map',
+    function.comp_app, continuous_linear_map.inl_apply, continuous_linear_map.prod_apply,
+    continuous_linear_map.add_apply, continuous_linear_map.coe_smul',
+    continuous_linear_map.coe_fst', pi.smul_apply, algebra.id.smul_eq_mul, mul_one,
+    continuous_linear_map.coe_snd', mul_zero, add_zero, continuous_linear_map.inr_apply, zero_add],
+  { rw [← basis.fin_two_prod_zero 𝕜, matrix.to_lin_self],
+    simp only [fin.sum_univ_two, matrix.cons_val_zero, basis.fin_two_prod_zero, prod.smul_mk,
+      algebra.id.smul_eq_mul, mul_one, mul_zero, basis.fin_two_prod_one, prod.mk_add_mk,
+      add_zero] },
+  { rw [← basis.fin_two_prod_zero 𝕜, matrix.to_lin_self],
+    simp only [fin.sum_univ_two, matrix.cons_val_zero, basis.fin_two_prod_zero, prod.smul_mk,
+      algebra.id.smul_eq_mul, mul_one, mul_zero, matrix.cons_val_one, matrix.head_cons,
+      basis.fin_two_prod_one, prod.mk_add_mk, zero_add] },
+  { rw [← basis.fin_two_prod_one 𝕜, matrix.to_lin_self],
+    simp only [fin.sum_univ_two, matrix.cons_val_zero, matrix.cons_val_one, matrix.head_cons,
+      basis.fin_two_prod_zero, prod.smul_mk, algebra.id.smul_eq_mul, mul_one, mul_zero,
+      basis.fin_two_prod_one, prod.mk_add_mk, add_zero] },
+  { rw [← basis.fin_two_prod_one 𝕜, matrix.to_lin_self],
+    simp only [fin.sum_univ_two, matrix.cons_val_one, matrix.head_cons,
+      basis.fin_two_prod_zero, prod.smul_mk, algebra.id.smul_eq_mul, mul_one, mul_zero,
+      basis.fin_two_prod_one, prod.mk_add_mk, zero_add] }
+end
```

Comment:
This feels much more pleasant to prove in the unbundled setting first:
```suggestion
lemma to_lin_fin_two_prod (a b c d : 𝕜) (x : 𝕜 × 𝕜) :
  matrix.to_lin (basis.fin_two_prod 𝕜) (basis.fin_two_prod 𝕜) ![![a, b], ![c, d]] x =
    (a * x.fst + b * x.snd, c * x.fst + d * x.snd) :=
begin
  rw [matrix.to_lin_apply, coe_fin_two_prod_repr, fin.sum_univ_two, basis.fin_two_prod_zero,
    basis.fin_two_prod_one, prod.smul_mk, prod.smul_mk, smul_zero, smul_zero, prod.mk_add_mk,
    zero_add, add_zero, smul_eq_mul, mul_one, smul_eq_mul, mul_one],
  -- missing some matrix lemmas for `fin n`
  sorry,
  -- simp?,
end

lemma to_lin_prod_continuous_linear_map (a b c d : 𝕜) :
  (matrix.to_lin (basis.fin_two_prod 𝕜) (basis.fin_two_prod 𝕜)
      ![![a, b], ![c, d]]).to_continuous_linear_map =
  (a • continuous_linear_map.fst 𝕜 𝕜 𝕜 + b • continuous_linear_map.snd 𝕜 𝕜 𝕜).prod
  (c • continuous_linear_map.fst 𝕜 𝕜 𝕜 + d • continuous_linear_map.snd 𝕜 𝕜 𝕜) :=
continuous_linear_map.ext $ to_lin_fin_two_prod _ _ _ _
```
where the first lemma belongs in an earlier file.